pkgs/tools/misc/kb/default.nix

```
@@ -5,14 +5,15 @@
 
 python3.pkgs.buildPythonApplication rec {
   pname = "kb";
-  version = "0.1.7";
+  version = "20231017-${lib.strings.substring 0 7 rev}";
```

```suggestion
  version = "0.1.7-unstable-2023-10-17";
```

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming